clients/unshaded/src/main/java/tachyon/client/LocalBlockOutStream.java

```diff
@@ -0,0 +1,254 @@
+
+  private final int mBlockIndex;
+  private final long mBlockCapacityByte;
+  private final long mBlockId;
+  private final long mBlockOffset;
+  private final Closer mCloser = Closer.create();
+  private final String mLocalFilePath;
+  private final RandomAccessFile mLocalFile;
+  private final FileChannel mLocalFileChannel;
+  private final ByteBuffer mBuffer;
+
+  private long mAvailableBytes = 0;
+  private long mInFileBytes = 0;
+  private long mWrittenBytes = 0;
+
+  private boolean mCanWrite = false;
+  private boolean mClosed = false;
+
+  /**
+   * @param file the file the block belongs to
+   * @param opType the OutStream's write type
+   * @param blockIndex the index of the block in the file
+   * @param tachyonConf the TachyonConf instance for this file output stream.
+   * @throws IOException
+   */
+  LocalBlockOutStream(TachyonFile file, WriteType opType, int blockIndex, TachyonConf tachyonConf)
+      throws IOException {
+    this(file, opType, blockIndex, tachyonConf.getBytes(Constants.USER_QUOTA_UNIT_BYTES,
+        8 * Constants.MB), tachyonConf);
+  }
+
+  /**
+   * @param file the file the block belongs to
+   * @param opType the OutStream's write type
+   * @param blockIndex the index of the block in the file
+   * @param initialBytes the initial size bytes that will be allocated to the block
+   * @param tachyonConf the TachyonConf instance for this file output stream.
+   * @throws IOException
+   */
+  LocalBlockOutStream(TachyonFile file, WriteType opType, int blockIndex, long initialBytes,
+                      TachyonConf tachyonConf) throws IOException {
+    super(file, opType, tachyonConf);
+
+    if (!opType.isCache()) {
+      throw new IOException("LocalBlockOutStream only supports WriteType.CACHE. opType: " + opType);
+    }
+
+    mBlockIndex = blockIndex;
+    mBlockCapacityByte = mFile.getBlockSizeByte();
+    mBlockId = mFile.getBlockId(mBlockIndex);
+    mBlockOffset = mBlockCapacityByte * blockIndex;
+
+    if (!mTachyonFS.hasLocalWorker()) {
+      throw new IOException("The machine does not have any local worker.");
+    }
+    mCanWrite = true;
+
+    // TODO: Use the new LocalFileBlockWriter api for writing local files.
+    mLocalFilePath = mTachyonFS.getLocalBlockTemporaryPath(mBlockId, initialBytes);
+    mLocalFile = mCloser.register(new RandomAccessFile(mLocalFilePath, "rw"));
+    mLocalFileChannel = mCloser.register(mLocalFile.getChannel());
+    // change the permission of the temporary file in order that the worker can move it.
+    CommonUtils.changeLocalFileToFullPermission(mLocalFilePath);
+    // use the sticky bit, only the client and the worker can write to the block
+    CommonUtils.setLocalFileStickyBit(mLocalFilePath);
+    LOG.info(mLocalFilePath + " was created!");
+    mAvailableBytes += initialBytes;
+
+    long allocateBytes = mTachyonConf.getBytes(Constants.USER_FILE_BUFFER_BYTES, Constants.MB) + 4L;
```

Comment:
Do we need the `+ 4L`? Also maybe make the `userFileBufferBytes` an instance variable since we use it multiple times (and the conf is presumably constant for now)?